features/netsocket/SocketAddress.cpp

```diff
@@ -215,17 +215,17 @@ void SocketAddress::set_port(uint16_t port)
 
 const char *SocketAddress::get_ip_address() const
 {
-    char *ip_address = (char *)_ip_address;
-
-    if (!ip_address[0]) {
+    if (!_ip_address[0]) {
         if (_addr.version == NSAPI_IPv4) {
-            ipv4_to_address(ip_address, _addr.bytes);
+            ipv4_to_address(_ip_address, _addr.bytes);
         } else if (_addr.version == NSAPI_IPv6) {
-            ipv6_to_address(ip_address, _addr.bytes);
+            ipv6_to_address(_ip_address, _addr.bytes);
+        } else {
```

Comment:
This doesn't make sense. "None/Unspec" in a socket address is not "0.0.0.0" - that could be inferred as a statement that it's IPv4.  Suggest "*", which is what you see in inetstat for unbound sockets.